R/draws.R

```diff
@@ -1,10 +1,97 @@
+#' from a MMRM model).
+#'
+#' The function distinguishes between the following methods:
+#' - Bayesian MI based on MCMC sampling: `draws` returns the draws
+#' from the posterior distribution of the parameters using a Bayesian approach based on
+#' MCMC sampling. This method can be specified by using `method = method_bayes()`.
+#' - Approximate Bayesian MI based on bootstrap: `draws` returns
+#' the draws from the posterior distribution of the parameters using a Bayesian approach,
+#' where the sampling from the posterior distribution is simulated by fitting the MMRM model
+#' on bootstrap samples of the original dataset. This method can be specified by using `method = method_approxbayes()]`.
+#' - Conditional mean imputation with bootstrap re-sampling: `draws` returns the
+#' MMRM parameter estimates from the original dataset and from `n_samples` bootstrap samples.
+#' This method can be specified by using `method = method_condmean()` with
+#' argument `type = "bootstrap"`.
+#' - Conditional mean imputation with Jackknife re-sampling: `draws` returns the
+#' MMRM parameter estimates from the original dataset and from each leave-one-out sample.
+#' This method can be specified by using `method = method_condmean()` with
+#' argument `type = "jackknife"`.
+#'
+#' The argument `data` must have one row per visit per subject. This means that incomplete
+#' outcome data must be set as `NA` instead of having the related row missing. Missing values
+#' in the covariates are not allowed. If `data` is incomplete
+#' then the [expand_locf()] helper function can be used to add in any missing rows, which uses
+#' Last Observation Carried Forward (LOCF) imputation to impute the covariates values.
+#'
+#' The argument `data_ice` contains information about the occurrence of ICEs. It is a
+#' `data.frame` with 3 columns:
+#' - A vector of characters related to the ids of the subjects that experienced
+#'   the ICE (must be named as `vars$subjid`).
+#' - A vector containing the first visit after the occurrence of the ICE (must be named as `vars$visit`).
+#'   This value must be equal to one of the levels of `data[[vars$visit]]`.
+#'   If multiple ICEs happen for the same subject, then only the first non-MAR visit should be used.
+#' - A vector of characters related to the imputation strategy to address the ICE (must be named as `vars$strategy`).
+#'   Possible imputation strategies are:
+#'   - `"MAR"`: Missing At Random.
+#'   - `"JTR"`: Jump To Reference.
+#'   - `"CIR"`: Copy Increments in Reference.
+#'   - `"CR"`: Copy Reference.
+#'   - `"JR"`: Jump to Reference.
+#'   - `"LMCF"`: Last Mean Carried Forward.
+#'   - User-defined imputation strategies can also be set.
+#'
+#' This argument is necessary at this stage since the model is fitted
+#' after removing the observations whose underlying assumption is different from MAR.
+#' Thus, any observed data after `data_ice[[vars$visit]]` that is addressed with an imputation
```

Comment:
should this  be   "after" -> "on or after"

Reply:
agreed